C cleanups

ronnychevalier opened the pull request:
Another batch of C cleanups

pkgw commented:
More great work. I am especially happy to see you tackle the sign-compare warnings!

My preferred style is to write out `unsigned int` instead of just `unsigned`. Could you please change those?

ronnychevalier commented:
@pkgw I changed every `unsigned` to `unsigned int` as requested.

About the sign-compare warnings I fixed many of them but there are still some remaining :)

pkgw commented:
Kickin' rad.
